### PR TITLE
Improve embedart logging management

### LINKED ISSUE #1244: embedart: Logging is too verbose in new infrastructure
The recent plugin log refactoring has made the `embedart` plugin too noisy. Since it uses an event handler rather than an import stage, its `self._log.info` calls (e.g., "compare PHASH score…") are printed during the import process. We need a way to quiet this down, either special-purpose for this plugin on its own.

A quick grep reveals that the plugins using `album_imported` and `item_imported` are: `embedart`, `freedesktop`, `importadded`, `importfeeds`, and `permissions`. All of them could use a second look to make sure they don't have the same problem.

I don't have any bright ideas at the moment, but I'll keep thinking. @brunal, any thoughts?

### PR DESCRIPTION
**Pull request by brunal**
- better logging levels
- always use the preferred user formatting for items & albums
- improve some messages wording

Also note there's small flow improvements, e.g. ex-lines 122-129.

Fix #1244

### CODE REVIEW

**Comment by brunal** [comment]
You'll see I added top-level functions to format item & album. I think that the current way is a bit too clumsy (find the template, then call `evaluate_template()`) and could be improved so it would feel natural for every logging statement to use it.

Formatting could then be a top-level option considered by all plugins.

**Comment by sampsyo** [comment]
Good point about global formatting options. Perhaps we could use [PEP 3101](https://www.python.org/dev/peps/pep-3101/) `__format__` methods to enable using `"{}".format(item)` directly.

File: beetsplug/embedart.py [hunk 1]
```diff
@@ -27,6 +26,12 @@
 from beets.util import syspath, normpath, displayable_path
 from beets.util.artresizer import ArtResizer
 from beets import config
+from beets.util.functemplate import Template
+
+__item_template = Template(ui._pick_format(False))
+fmt_item = lambda item: item.evaluate_template(__item_template)
+__album_template = Template(ui._pick_format(True))
+fmt_album = lambda item: item.evaluate_template(__album_template)
```
Comment (sampsyo) [inline beetsplug/embedart.py]: You're right that it's a little unfortunate that we're calling `ui._pick_format` on module initialization. That kind of thing can get brittle. Let's execute your idea about object formatting Real Soon Now™.

File: beetsplug/embedart.py [hunk 2]
```diff
@@ -117,15 +123,10 @@ def embed_item(self, item, imagepath, maxwidth=None, itempath=None,
         if compare_threshold:
             if not self.check_art_similarity(item, imagepath,
                                              compare_threshold):
-                self._log.warn(u'Image not similar; skipping.')
+                self._log.debug(u'Image not similar; skipping.')
```
Comment (sampsyo) [inline beetsplug/embedart.py]: I'd lean `info` on this one over `debug`: you probably would want to know this when running `beet fetchart` directly.

**Comment by sampsyo** [comment]
This looks great—the code changes are useful and the level changes are definitely needed.

Does this really fix #1244, though? In the end, we want lines like "Embedding album art into…" to be printed when running `beet fetchart` but *not* in import mode. Since fetchart uses an event and not a stage, `info` lines are always printed (right?).

**Comment by brunal** [comment]
About formatting: implementing `__format__` on items and albums would be really convenient\*, however that would require them to know about `ui` which should be avoided, shouldn't it? Possible solutions include
* Not caring about that and importing `beets.ui` inside `Item.__format__` and `Album.__format__`
* Monkey-patching those methods on ui initialization (ugly for they are private methods....)
* Providing functions `beets.ui.fmt_item` and `beets.ui.fmt_album` (less automatic = less cool)
* Wrapping `Item` & `Album` with child classes `UiItem` and `UiAlbum` (may be messy…)

About log levels: that seems a bit tricky. When the `EmbedCoverArtPlugin` is called because of `fetchart` we'd like to display info lines but when it is called by `import` we don't? Therefore the `fetchart` plugin should change `embedart`'s logger level in `batch_fetch_art`?

\* we could do that with paths as well, couldn't we? `__format__` = `displayable_path`

File: beetsplug/embedart.py (hunk 2 above)
Comment (brunal) [inline beetsplug/embedart.py]: Alright, then the same would apply to the next logging statement, `media file already contained art`, wouldn't it?

**Comment by brunal** [comment]
Other (simpler) solution: `format` is a class attribute for both `Item` and `Album` and gets set upon beets ui initialization. That attribute is then used in the `__format__()` methods, if set. That's the best solution I can see: it does not break  the abstraction layers (though the leak a bit), it's not hackish and logging would be natural.

**Comment by sampsyo** [comment]
What about just moving the `_pick_format` nonsense out of `beets.ui`? That is, the `__format__` methods in Library would just use `beets.config['list_format_item']` and `beets.config['list_format_album']` directly. Those particular configuration names may be obsolete, but I'm not concerned about using the formatting configuration in Library. (You're right that importing the ui module would be troublesome.)

File: beetsplug/embedart.py (hunk 2 above)
Comment (sampsyo) [inline beetsplug/embedart.py]: Yes, good point.

Comment (brunal) [inline beetsplug/embedart.py]: Done!

**Comment by sampsyo** [comment]
Merged (thanks!!), but I re-opened #1244. Still need to do some investigation to get a clean solution to level-shifting (and audit other import plugins for similar behavior).

**Pull request merged**